scripts/create-component/create-component.ts

```diff
@@ -0,0 +1,271 @@
+  if (fs.existsSync(location) !== true) {
+    displayAndThrowError(
+      `**ABORTING** The package ${packageName} cannot be found at location ${location}. Use yarn create-package first.`,
+    );
+  }
+  if (fs.existsSync(`${location}/src/index.ts`) !== true) {
+    displayAndThrowError(`**ABORTING** The package ${packageName} was found but missing src/index.ts`);
+  }
+
+  return `Found package ${packageName} at location: ${rootPaths.package}`;
+};
+
+const checkIfComponentAlreadyExists = (answers: object, config: object, plop: object): string => {
+  const { componentName } = answers as Answers;
+  const plopAPI = plop as NodePlopAPI;
+  const location = plopAPI.renderString(rootPaths.component, answers);
+
+  if (fs.existsSync(location) === true && fs.readdirSync(location).length > 0) {
+    displayAndThrowError(`**ABORTING** The component ${componentName} already exists at ${location}`);
+  }
+  return `Component ${componentName} doesn't exist.`;
+};
+
+const appendToPackageIndex = async (answers: object, config: object, plop: object): Promise<string> => {
+  const { componentName, packageName } = answers as Answers;
+  const plopAPI = plop as NodePlopAPI;
+
+  const options = { flag: 'a' };
+  // get the package index file path
+  const indexPath = plopAPI.renderString(`${rootPaths.package}/src/index.ts`, answers);
+  const appendLine = plopAPI.renderString(`export * from './{{componentName}}';`, answers);
+  // read contents and see if line is exists
+  return fs
+    .readFile(indexPath, { encoding: 'utf8', flag: 'r' })
+    .then(async data => {
+      const lines = data.split(/\r?\n/);
+      const getIndex = (arr: string[], item: string): number =>
+        lines.findIndex(line => item.toLocaleLowerCase() === line.toLocaleLowerCase());
+      if (getIndex(lines, appendLine) === -1) {
+        // doesn't exist so append
+        await fs.outputFile(indexPath, `${appendLine}${os.EOL}`, options);
+        return `Updated package ${packageName} index.ts to include ${componentName}`;
+      }
+      return `Package ${packageName} index.ts already contains reference to ${componentName}`;
+    })
+    .catch(error => {
+      throw `**ABORTING** There was an error reading index file at ${indexPath}. Error: ${error}`;
+    });
+};
+//#endregion
+
+//#region Utilities
+const displayAndThrowError = (message: string) => {
+  console.log(message);
+  throw message;
+};
+
+//#endregion
+
+const ignoreProjects = [
+  '@fluentui/azure-themes',
+  '@fluentui/docs',
+  '@fluentui/dom-utilities',
+  '@fluentui/react-examples',
+  '@fluentui/foundation-legacy',
+  '@fluentui/react-next',
+  '@fluentui/react-next',
+  '@fluentui/react',
+  '@fluentui/test-utilities',
+  '@fluentui/theme',
+  '@fluentui/webpack-utilities',
+];
+const projectsWithStartCommand = Object.entries(allPackages)
+  .reduce((acc, [pkg, info]) => {
+    if (info.packageJson.scripts && info.packageJson.scripts.start && info.packagePath.startsWith('packages')) {
```

Comment:
I feel that using reduce in this context is much more clearer here and if you wish, can we have a discussion with suggested changes and/or move this to a future PR?

Reply:
I dislike `reduce` in most cases because it can be used in so many different ways and therefore introduces an additional level of mental gymnastics which (at least to me) makes it harder to follow what the code is doing.

```
Object.entries(allPackages)
  .filter(([pkg, info]) => !ignoreProjects.includes(pkg) && info.packagePath.startsWith('packages')
    && !!info.packageJson.scripts?.start)
  .map(([pkg, info]) => ({ title: pkg, value: { pkg, command: 'start' } }))
```

Reply:
Updated